### ISSUE TITLE
Got "Cannot read property 'url' of undefined" error when using .renderRoute

### ISSUE THREAD

**Issue by f15gdsy** [nuxt/nuxt.js#426]
Hi, I went into trouble when using renderRoute.
The following is what I did:

## Env
Mac OS X EI Capitan 10.11.6
Node v7.7.4

## 5 Simple Steps
1. Download nuxt-koa template
```bash
$ vue init nuxt/koa (choose koa v2)
```

2. Install dependencies
```bash
$ yarn
```

3. Try to run before changing anything, and it works fine
```bash
$ npm run dev  # which works fine
```

4. Change app.js to use renderRoute
```javascript
// app.js

// ...

// Change the middleware to use renderRoute
app.use(async (ctx, next) => {
  ctx.status = 200 // koa defaults to 404 when it sees that status is unset
  // await nuxt.render(ctx.req, ctx.res)
  const result = await nuxt.renderRoute('/')
  console.log(result.error)
})
```

5. Run again
```bash
$ npm run dev # error is printed
```

Error Message:
```
{ statusCode: 500,
  message: 'Cannot read property \'url\' of undefined' }
```

Am I using renderRoute in a wrong way?
Any idea?
Thanks!

**Comment by f15gdsy** [comment]
Any ideas? Got stuck in this problem..

**Comment by Atinux** [comment]
Hi @f15gdsy 

I will be fixed in the `0.10` release, see https://github.com/nuxt/nuxt.js/commit/26083a9de0be8f748456d1b2d8fbdf0529881536

But anyway, you should give `req` & `res` to the `renderRoute` method:
```js
app.use(async (ctx, next) => {
  ctx.status = 200 // koa defaults to 404 when it sees that status is unset
  const result = await nuxt.renderRoute('/', {
    req: ctx.req,
    res: ctx.res
  })
  console.log(result.error)
})
```

**Comment by f15gdsy** [comment]
Thanks!
